torchtune/datasets/_slimorca.py

```diff
@@ -4,53 +4,54 @@
-    The Llama3 models do not prescribe a particular format.
-
-    The returned data is a tuple of input token id list and label token id
-    list. If ``max_seq_len`` keyword argument is provided, the returned
-    input token id list is ensured (by truncation if necessary) to be within
-    that length.
+    Masking of the prompt during training is controlled by the ``train_on_input`` flag, which is
+    set to ``False`` by default
+    - If ``train_on_input`` is True, the prompt is used during training and
+    contributes to the loss.
+    - If ``train_on_input`` is False, the prompt is masked out (tokens replaced with -100)
 
     Args:
-        tokenizer (ModelTokenizer): Tokenizer used by the model that implements the ``tokenize_messages`` method.
-        source (str): path string of dataset, anything supported by Hugging Face's `load_dataset`.
-        chat_format (Optional[str]): name of template used to format the chat. See the description
-            in :class:`~torchtune.datasets.ChatDataset` for more details. Default: None
-        max_seq_len (int): Maximum number of tokens in the returned input and label token id lists.
-            This value needs to be at least 4 though it is generally set to max sequence length accepted by the model.
-            Default is 1024.
+        model_transform (Transform): model specific transform to convert a list of messages
+            output by the dataset to tokens. This will always be a :class:`~torchtune.modules.tokenizers.ModelTokenizer`.
+        source (str): path to dataset repository on Hugging Face. For local datasets,
+            define source as the data file type (e.g. "json", "csv", "text") and pass
+            in the filepath in ``data_files``. See `Hugging Face's
+            <https://huggingface.co/docs/datasets/en/package_reference/loading_methods#datasets.load_dataset.path>`_
+            ``load_dataset`` for more details. Default is ``Open-Orca/SlimOrca-Dedup``.
+        column_map (Optional[Dict[str, str]]): a mapping from the expected columns in the prompt template
+            to the new column names in the dataset. If None, assume these are identical.
+        prompt_template (Optional[PromptTemplate]): optional template used to format the prompt. Default
+            is None.
         train_on_input (bool): Whether the model is trained on the prompt or not. Default is False.
         packed (bool): Whether or not to pack the dataset to ``max_seq_len`` prior to training. Default is False.
         split (str): ``split`` argument for ``datasets.load_dataset``. You can use this argument to load a subset
             of a given split, e.g. ``split="train[:10%]"``. Default is "train".
 
-    Raises:
-        ValueError: If `max_seq_len` is less than 4.
-
     Returns:
-        ChatDataset: dataset configured with SlimOrca source data and Llama2 chat template
+        Union[SFTDataset, PackedDataset]: dataset configured with SlimOrca source data
```

Comment:
Can we use the `|` operator yet for this? Or is that only available in Python 3.11?

Reply:
Like `SFTDataset | PackedDataset`

Reply:
3.10 and above, and I think we test 3.9?

Reply:
Grrr okay. Soon, soon.